### DISCUSSION TITLE
Get assembly files?

### DISCUSSION THREAD

**Discussion by chriselrod** [mesonbuild/meson#10863] [Ideas]
It'd be very useful if we could get assembly files (and to also be able to set `-masm=intel` on x64).
However, I could not find any way of doing this.

Googling suggested adding `-save-temps` as an option should work, but I cannot successfully compile with it.
```
/usr/bin/../lib/gcc/x86_64-redhat-linux/12/../../../../include/c++/12/x86_64-redhat-linux/bits/c++config.h:2549:15: error: redefinition of '__terminate'
  inline void __terminate() noexcept
```

**Comment by LunarLambda** [comment]
If you only need this for a few files from your build, I think you can get the compiler with `meson.get_compiler(...)` and then pass it to `custom_target` or similar to generate the assembly as needed (i.e. `gcc -masm=intel -S -o asm.s source.cpp`)

GCC also lets you control where the temporaries are stored, using `-dumpbase`/`-dumpdir` and `-save-temps=...`. https://gcc.gnu.org/onlinedocs/gcc/Overall-Options.html#Overall-Options, that might help?
